src/pulse3D/excel_writer.py

```diff
@@ -221,6 +224,79 @@ def write_xlsx(
     interpolated_data_period = (
         w[INTERPOLATION_VALUE_UUID] if plate_recording.is_optical_recording else INTERPOLATED_DATA_PERIOD_US
     )
+
+    # get stim meta data
+    if include_stim_protocols:
+        if w[STIMULATION_PROTOCOL_UUID] == "null":
+            stim_protocols_df = pd.DataFrame({"No stimulation protocols have been used for this recording"})
+        else:
+            protocols_id_list = []
+            unassigned_wells = ""
+            stim_protocols_dict = {
+                "Title": {
+                    "Unassigned Wells": "Unassigned Wells:",
+                    "title_break": "",
+                    "Protocol ID": "Protocol ID:",
+                    "Stimulation Type": "Stimulation Type:",
+                    "Run Until Stopped": "Run Until Stopped:",
+                    "Wells": "Wells:",
+                    "Subprotocols": "Subprotocols:",
+                }
+            }
+
+            for well in plate_recording.wells:
+                well_data = json.loads(well[STIMULATION_PROTOCOL_UUID])
+
+                if well_data is not None:
+                    protocol_id = well_data.get("protocol_id")
+                    well_id = well[WELL_NAME_UUID]
+
+                    if protocol_id not in protocols_id_list:
+                        protocols_id_list.append(protocol_id)
+
+                        stim_protocols_dict[protocol_id] = {
+                            "Protocol ID": protocol_id,
+                            "Stimulation Type": "Current"
+                            if well_data.get("stimulation_type") == "C"
+                            else "Voltage"
+                            if well_data.get("stimulation_type") == "V"
+                            else well_data.get("stimulation_type"),
+                            "Run Until Stopped": "Active"
+                            if well_data.get("run_until_stopped")
+                            else "Disabled",
+                            "Wells": f"{well_id}, ",
+                            "Subprotocols": "",
+                        }
+                        for i in range(len(well_data.get("subprotocols"))):
```

Comment:
The subprotocol structure is going to change in an update coming very soon, and this formatting won't work with the new structure. It will become even harder to do something like this once we need support looping and be backwards compatible with older protocols. Can we just try outputting subprotocol using json.dumps with the indent param given? That will support any format with minimal work

Reply:
When adding `indent`  the JSON is printed to one cell, but the indentation is lost. `indent` becomes a variable to control spaces. Most of the solution mentioned in https://stackoverflow.com/questions/12943819/how-to-prettyprint-a-json-file are for printing to screen. They do not work the same for excel sheets.